libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -50,6 +50,50 @@ string IRGenerationContext::localVariableName(VariableDeclaration const& _varDec
 	return m_localVariables[&_varDecl];
 }
 
+vector<string> IRGenerationContext::addCalldataVariable(VariableDeclaration const& _varDecl)
```

Comment:
Shouldn't this just be a special case of `addLocalVariable`? Local variables of storage pointer type are also not registered via `addStateVariable`.

Reply:
And then the variable is "called" `vloc_<name>_<id>`, but its "components" are `vloc_<name>_<id>_offset` and `vloca_<name>_<id>_length`, so we use `localVariableName(var, "offset")` and `localVariableName(var, "length")` to access the two components.

Reply:
So you do not even have to change `m_localVariables`, but instead these only provide the prefixes.

Reply:
Haha, yeah, I started that way. But then I'd need to change local variables to support multiple stack slots and then ``IRLValue`` as well and then all its derived classes. At least as long as we don't have tuple assignments :-). (With tuple assignments the variable name of the local variable could just be a comma-separated list, but still a single ``std::string``...)

Reply:
Can we talk about this in person tomorrow maybe?

Reply:
Yeah, probably makes sense.

Reply:
But yeah - once we have stack slot name information associated with the types, something like that might actually work nicely (and for some reason github only now displays your second and third comment to me).